### ISSUE TITLE
CRC 1.33.1 It is getting failed to start the crc

### ISSUE THREAD

**Issue by rupeshad81** [code-ready/crc#2783]
### General information

  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / Hyper-V / hyperkit
  * Did you run `crc setup` before starting it (Yes/No)?
  * Running CRC on: Laptop / Baremetal-Server / VM

## CRC version
```bash
# Put `crc version` output here
[crc@eaas-crc ~]$ crc version
CodeReady Containers version: 1.33.1+3cfaa10f
OpenShift version: 4.8.12 (embedded in executable)

  
## CRC status
```bash
# Put `crc status --log-level debug` output here
```

## CRC config
```bash
# Put `crc config view` output here
[crc@eaas-crc ~]$ crc status --log-level debug
DEBU CodeReady Containers version: 1.33.1+3cfaa10f
DEBU OpenShift version: 4.8.12 (embedded in executable)
DEBU Running 'crc status'
DEBU Checking file: /home/crc/.crc/machines/crc/.crc-exist
DEBU Checking file: /home/crc/.crc/machines/crc/.crc-exist
DEBU Found binary path at /home/crc/.crc/bin/crc-driver-libvirt
DEBU Launching plugin server for driver libvirt
DEBU Plugin server listening at address 127.0.0.1:38079
DEBU () Calling .GetVersion
DEBU Using API Version 1
DEBU () Calling .SetConfigRaw
DEBU () Calling .GetMachineName
DEBU (crc) Calling .GetState
DEBU (crc) DBG | time="2021-10-04T10:57:23+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-04T10:57:23+01:00" level=debug msg="Fetching VM..."
DEBU (crc) Calling .GetBundleName
DEBU (crc) Calling .GetIP
DEBU (crc) DBG | time="2021-10-04T10:57:23+01:00" level=debug msg="GetIP called for crc"
DEBU (crc) DBG | time="2021-10-04T10:57:23+01:00" level=debug msg="Getting current state..."
DEBU (crc) DBG | time="2021-10-04T10:57:23+01:00" level=debug msg="IP address: 192.168.130.11"
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1
DEBU Using ssh private keys: [/home/crc/.crc/machines/crc/id_ecdsa /home/crc/.crc/cache/crc_libvirt_4.8.12/id_ecdsa_crc]
DEBU SSH command results: err: <nil>, output: 32737570816 15077834752 /sysroot
DEBU Unexpected operator status for etcd: RecentBackup
DEBU image-registry operator is degraded, Reason: ImageRegistryCertificatesControllerError
DEBU Making call to close driver server
DEBU (crc) Calling .Close
DEBU (crc) DBG | time="2021-10-04T10:57:23+01:00" level=debug msg="Closing plugin on server side"
DEBU Successfully made call to close driver server
DEBU Making call to close connection to plugin binary
CRC VM:          Running
OpenShift:       Degraded (v4.8.12)
Disk Usage:      15.08GB of 32.74GB (Inside the CRC VM)
Cache Usage:     16.06GB
Cache Directory: /home/crc/.crc/cache


## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
```

### Steps to reproduce

  1. 
  2. 
  3. 
  4. 

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug

[crc@eaas-crc ~]$ crc start --cpus 12 --memory 40000
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
INFO Checking if crc executable symlink exists
INFO Checking if Virtualization is enabled
INFO Checking if KVM is enabled
INFO Checking if libvirt is installed
INFO Checking if user is part of libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
INFO Checking if a supported libvirt version is installed
INFO Checking if crc-driver-libvirt is installed
INFO Checking if systemd-networkd is running
INFO Checking if NetworkManager is installed
INFO Checking if NetworkManager service is running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
INFO Checking if libvirt 'crc' network is available
INFO Checking if libvirt 'crc' network is active
INFO Starting CodeReady Containers VM for OpenShift 4.8.12...
INFO CodeReady Containers instance is running with IP 192.168.130.11
INFO CodeReady Containers VM is running
INFO Check internal and public DNS query...
INFO Check DNS query from host...
INFO Verifying validity of the kubelet certificates...
INFO Starting OpenShift kubelet service
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Waiting for user's pull secret part of instance disk...
INFO Adding proxy configuration to the cluster...
INFO Starting OpenShift cluster... [waiting for the cluster to stabilize]
INFO Operator dns is progressing
INFO Operator image-registry is degraded
INFO Operator authentication is progressing
INFO 2 operators are progressing: authentication, console
INFO 2 operators are progressing: authentication, console
INFO 2 operators are progressing: authentication, console
INFO 2 operators are progressing: authentication, console
INFO 2 operators are progressing: authentication, console
INFO 2 operators are progressing: authentication, console
INFO Operator console is progressing
INFO Operator console is progressing
INFO Operator console is progressing
INFO Operator console is progressing
INFO Operator console is progressing
INFO Operator console is progressing
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
INFO Operator image-registry is degraded
ERRO Cluster is not ready: cluster operators are still not stable after 15m1.35155957s
INFO Waiting for the proxy configuration to be applied...
INFO Adding crc-admin and crc-developer contexts to kubeconfig...
Started the OpenShift cluster.

The server is accessible via web console at:
  https://console-openshift-console.apps-crc.testing

Log in as administrator:
  Username: kubeadmin
  Password: SJwN6-zgJWu-JqCGk-KDoKw

Log in as user:
  Username: developer
  Password: developer

Use the 'oc' command line interface:
  $ eval $(crc oc-env)
  $ oc login -u developer https://api.crc.testing:6443


Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

**Comment by gbraad** [comment]
According to:

```
Started the OpenShift cluster.

The server is accessible via web console at:
  https://console-openshift-console.apps-crc.testing

Log in as administrator:
  Username: kubeadmin
  Password: SJwN6-zgJWu-JqCGk-KDoKw

Log in as user:
  Username: developer
  Password: developer

Use the 'oc' command line interface:
  $ eval $(crc oc-env)
  $ oc login -u developer https://api.crc.testing:6443
```

The cluster started. I do not see the error as the title suggests: `It is getting failed to start the crc`. Can you clarify this?

**Comment by rupeshad81** [comment]
Hi,

Below status should show "Running"


 but it is showing "Degarded"

[crc@eaas-crc ~]$ crc status
CRC VM:          Running
OpenShift:       Degraded (v4.8.12)
Disk Usage:      15.08GB of 32.74GB (Inside the CRC VM)
Cache Usage:     16.06GB
Cache Directory: /home/crc/.crc/cache

**Comment by gbraad** [comment]
Did you try to access the cluster?

---

```
ERRO Cluster is not ready: cluster operators are still not stable after 15m1.35155957s
```

as this is only showing `INFO Operator image-registry is degraded`.

**Comment by rupeshad81** [comment]
Hi,

Yes this is also showing.

**Comment by gbraad** [comment]
Can you access the cluster on the specified address?

And can you share `crc config view` ?

**Comment by rupeshad81** [comment]
Hi,

I am able to open the console .

[crc@eaas-crc ~]$ crc config view
- consent-telemetry                     : yes
- http-proxy                            : http://10.232.233.70:8080
- https-proxy                           : http://10.232.233.70:8080
- no-proxy                              : .apps.ilocpfnd401.ocp.corp.amdocs.com,.cluster.local,.ilocpfnd401.ocp.corp.amdocs.com,.svc,10.0.0.0/16,10.254.0.0/16,127.0.0.1,172.30.0.0/16,api-int.ilocpfnd401.ocp.corp.amdocs.com,api.ilocpfnd401.ocp.corp.amdocs.com,etcd-0.ilocpfnd401.ocp.corp.amdocs.com,etcd-1.ilocpfnd401.ocp.corp.amdocs.com,etcd-2.ilocpfnd401.ocp.corp.amdocs.com,ilvcos01.corp.amdocs.com,localhost,.corp.amdocs.com,.apps-crc.testing,.crc.testing
![CRC_1_33_1](https://user-images.githubusercontent.com/39945481/135833414-1c6b10d6-e1e0-4b23-aab0-644093886b39.JPG)

**Comment by rupeshad81** [comment]
Hi,

But in the crc status, Openshift is showing " Degarded"

[crc@eaas-crc ~]$ crc status
CRC VM:          Running
OpenShift:       Degraded (v4.8.12)
Disk Usage:      15.15GB of 32.74GB (Inside the CRC VM)
Cache Usage:     16.06GB
Cache Directory: /home/crc/.crc/cache

**Comment by praveenkumar** [comment]
@rupeshad81 Your proxy is using tls certificate and I don't see you are specify it as part of `crc config set proxy-ca-file` so first do that and then try again.

**Comment by gbraad** [comment]
the cluster works, but you get a `DEBU image-registry operator is degraded, Reason: ImageRegistryCertificatesControllerError` so likely you use a CA-signed (company signed) certificate that causes the issues

**Comment by rupeshad81** [comment]
Hi Praveen,

I had used the below command to set:

crc config set http-proxy http://10.232.233.70:8080
crc config set https-proxy http://10.232.233.70:8080
crc config set no-proxy .apps.ilocpfnd401.ocp.corp.amdocs.com,.cluster.local,.ilocpfnd401.ocp.corp.amdocs.com,.svc,10.0.0.0/16,10.254.0.0/16,127.0.0.1,172.30.0.0/16,api-int.ilocpfnd401.ocp.corp.amdocs.com,api.ilocpfnd401.ocp.corp.amdocs.com,etcd-0.ilocpfnd401.ocp.corp.amdocs.com,etcd-1.ilocpfnd401.ocp.corp.amdocs.com,etcd-2.ilocpfnd401.ocp.corp.amdocs.com,ilvcos01.corp.amdocs.com,localhost,.corp.amdocs.com,.apps-crc.testing,.crc.testing


Let me know, what other need to set ?


Thanks

**Comment by praveenkumar** [comment]
You need to set the proxy ca also like https://github.com/code-ready/crc/issues/2783#issuecomment-933340092 mentioned here.

**Comment by gbraad** [comment]
Please read the comments carefully as it has been mentioned twice to do the following:

```
$ crc config set proxy-ca-file [cert-file-location
```

**Comment by rupeshad81** [comment]
Hi,

To set this certificate :

crc config set proxy-ca-file /tmp/amdocs-certificat.crt

And then do the installation.

**Comment by rupeshad81** [comment]
Hi,

Please do not close the issue, till it get successfully passed.


Thanks

**Comment by rupeshad81** [comment]
Hi,

Steps for installation will be below:


crc config set http-proxy http://10.232.233.70:8080
crc config set https-proxy http://10.232.233.70:8080
crc config set no-proxy .apps.ilocpfnd401.ocp.corp.amdocs.com,.cluster.local,.ilocpfnd401.ocp.corp.amdocs.com,.svc,10.0.0.0/16,10.254.0.0/16,127.0.0.1,172.30.0.0/16,api-int.ilocpfnd401.ocp.corp.amdocs.com,api.ilocpfnd401.ocp.corp.amdocs.com,etcd-0.ilocpfnd401.ocp.corp.amdocs.com,etcd-1.ilocpfnd401.ocp.corp.amdocs.com,etcd-2.ilocpfnd401.ocp.corp.amdocs.com,ilvcos01.corp.amdocs.com,localhost,.corp.amdocs.com,.apps-crc.testing,.crc.testing
crc config set proxy-ca-file /tmp/amdocs-certificat.crt


crc setup

crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json

**Comment by praveenkumar** [comment]
@rupeshad81 after setting `crc config set proxy-ca-file /tmp/amdocs-certificat.crt`  have you tried `crc delete` and `crc start`?

**Comment by rupeshad81** [comment]
Hi,

Please can we have call setup to get more clear about the steps, so that we will for follow the same steps for future also.

Thanks

**Comment by rupeshad81** [comment]
Hi,

Below status :

[crc@eaas-crc ~]$ crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
INFO Checking if crc executable symlink exists
INFO Checking if Virtualization is enabled
INFO Checking if KVM is enabled
INFO Checking if libvirt is installed
INFO Checking if user is part of libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
INFO Checking if a supported libvirt version is installed
INFO Checking if crc-driver-libvirt is installed
INFO Checking if systemd-networkd is running
INFO Checking if NetworkManager is installed
INFO Checking if NetworkManager service is running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
INFO Checking if libvirt 'crc' network is available
INFO Checking if libvirt 'crc' network is active
INFO Loading bundle: crc_libvirt_4.8.12...
INFO Creating CodeReady Containers VM for OpenShift 4.8.12...
INFO Generating new SSH Key pair...
INFO Generating new password for the kubeadmin user
INFO Starting CodeReady Containers VM for OpenShift 4.8.12...
INFO CodeReady Containers instance is running with IP 192.168.130.11
INFO CodeReady Containers VM is running
INFO Updating authorized keys...
INFO Check internal and public DNS query...
INFO Check DNS query from host...
INFO Verifying validity of the kubelet certificates...
INFO Starting OpenShift kubelet service
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Adding user's pull secret to the cluster...
INFO Updating SSH key to machine config resource...
Failed to update ssh public key to machine config: Failed to update ssh key ssh command error:
command : timeout 30s oc patch machineconfig 99-master-ssh -p '{"spec": {"config": {"passwd": {"users": [{"name": "core", "sshAuthorizedKeys": ["ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1MjEAAACFBAEYAK0y+WOOWhM1kqp/Y6wgq7QxXM3t7/lCLfhI2pu29XbChJG1dQENK3pKvZ+jCwFfof+PvSmuic4vEku8/Vo98ADXWBrm0/0cCo8KnehOGY01HfwQuuVHcWUFmP+ce+XnL2mrvtWBsfh1diW5DiWLNCq37SxHcblrEBDQ1vb/hVvxRg=="]}]}}}}' --type merge --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : wait: remote command exited without exit status or exit signal

**Comment by gbraad** [comment]
> to get more clear about the steps, so that we will for follow the same steps for future also.

Please read the documentation: https://crc.dev/crc/#starting-codeready-containers-behind-proxy_gsg and follow the instructions as asked for.

> Please can we have call setup

We aren't a support help desk. We will only set up a call when we really need to debug an issue we can not easily reproduce. So far we not at that stage.

**Comment by rupeshad81** [comment]
Hi,

But get the more clear about the steps to run , as it is showing the below failures.

[crc@eaas-crc ~]$ crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
INFO Checking if crc executable symlink exists
INFO Checking if Virtualization is enabled
INFO Checking if KVM is enabled
INFO Checking if libvirt is installed
INFO Checking if user is part of libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
INFO Checking if a supported libvirt version is installed
INFO Checking if crc-driver-libvirt is installed
INFO Checking if systemd-networkd is running
INFO Checking if NetworkManager is installed
INFO Checking if NetworkManager service is running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
INFO Checking if libvirt 'crc' network is available
INFO Checking if libvirt 'crc' network is active
INFO Loading bundle: crc_libvirt_4.8.12...
INFO Creating CodeReady Containers VM for OpenShift 4.8.12...
INFO Generating new SSH Key pair...
INFO Generating new password for the kubeadmin user
INFO Starting CodeReady Containers VM for OpenShift 4.8.12...
INFO CodeReady Containers instance is running with IP 192.168.130.11
INFO CodeReady Containers VM is running
INFO Updating authorized keys...
INFO Check internal and public DNS query...
INFO Check DNS query from host...
INFO Verifying validity of the kubelet certificates...
INFO Starting OpenShift kubelet service
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Adding user's pull secret to the cluster...
INFO Updating SSH key to machine config resource...
Failed to update ssh public key to machine config: Failed to update ssh key ssh command error:
command : timeout 30s oc patch machineconfig 99-master-ssh -p '{"spec": {"config": {"passwd": {"users": [{"name": "core", "sshAuthorizedKeys": ["ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1MjEAAACFBAEYAK0y+WOOWhM1kqp/Y6wgq7QxXM3t7/lCLfhI2pu29XbChJG1dQENK3pKvZ+jCwFfof+PvSmuic4vEku8/Vo98ADXWBrm0/0cCo8KnehOGY01HfwQuuVHcWUFmP+ce+XnL2mrvtWBsfh1diW5DiWLNCq37SxHcblrEBDQ1vb/hVvxRg=="]}]}}}}' --type merge --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : wait: remote command exited without exit status or exit signal

**Comment by gbraad** [comment]
The steps you performed:

```
crc config set proxy-ca-file /tmp/amdocs-certificat.crt
crc setup
crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json
```

are not as expected. The `setup` command does not need to be re-run. After the `config` change you have:

```
crc delete
crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json
```

**Comment by rupeshad81** [comment]
Hi,

No, I had the below steps :


crc config set proxy-ca-file /tmp/amdocs-certificat.crt

crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json


Please check below 


[crc@eaas-crc ~]$ crc config set proxy-ca-file /tmp/amdocs-certificat.crt
Successfully configured proxy-ca-file to /tmp/amdocs-certificat.crt
[crc@eaas-crc ~]$ crc delete
Do you want to delete the OpenShift cluster? [y/N]: y
Deleted the OpenShift cluster
[crc@eaas-crc ~]$ crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
INFO Checking if crc executable symlink exists
INFO Checking if Virtualization is enabled
INFO Checking if KVM is enabled
INFO Checking if libvirt is installed
INFO Checking if user is part of libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
INFO Checking if a supported libvirt version is installed
INFO Checking if crc-driver-libvirt is installed
INFO Checking if systemd-networkd is running
INFO Checking if NetworkManager is installed
INFO Checking if NetworkManager service is running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
INFO Checking if libvirt 'crc' network is available
INFO Checking if libvirt 'crc' network is active
INFO Loading bundle: crc_libvirt_4.8.12...
INFO Creating CodeReady Containers VM for OpenShift 4.8.12...
INFO Generating new SSH Key pair...
INFO Generating new password for the kubeadmin user
INFO Starting CodeReady Containers VM for OpenShift 4.8.12...
INFO CodeReady Containers instance is running with IP 192.168.130.11
INFO CodeReady Containers VM is running
INFO Updating authorized keys...
INFO Check internal and public DNS query...
INFO Check DNS query from host...
INFO Verifying validity of the kubelet certificates...
INFO Starting OpenShift kubelet service
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Adding user's pull secret to the cluster...
INFO Updating SSH key to machine config resource...
Failed to update ssh public key to machine config: Failed to update ssh key ssh command error:
command : timeout 30s oc patch machineconfig 99-master-ssh -p '{"spec": {"config": {"passwd": {"users": [{"name": "core", "sshAuthorizedKeys": ["ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1MjEAAACFBAEYAK0y+WOOWhM1kqp/Y6wgq7QxXM3t7/lCLfhI2pu29XbChJG1dQENK3pKvZ+jCwFfof+PvSmuic4vEku8/Vo98ADXWBrm0/0cCo8KnehOGY01HfwQuuVHcWUFmP+ce+XnL2mrvtWBsfh1diW5DiWLNCq37SxHcblrEBDQ1vb/hVvxRg=="]}]}}}}' --type merge --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : wait: remote command exited without exit status or exit signal
:

**Comment by gbraad** [comment]
> No, I had the below steps :

And that is your problem. At no point do I se you perform the step we asked you to do. After a config change, like CPU, memory, pull-secret, proxy, etc, you need to remove the current VM and start again.

I do not see you perform the `crc delete`.

**Comment by rupeshad81** [comment]
Hi,

Ok , performing the same steps.

Will update you.

Please do not close the issue, till it is passed.

Thanks

**Comment by gbraad** [comment]
> Please do not close the issue, till it is passed.

If we have any indication the issue is not related to CRC we will close the issue.

**Comment by rupeshad81** [comment]
Hi,


please check below:


[crc@eaas-crc ~]$ crc delete
Do you want to delete the OpenShift cluster? [y/N]: y
Deleted the OpenShift cluster
[crc@eaas-crc ~]$ crc start --cpus 12 --memory 40000 --pull-secret-file /tmp/pullsecret.json
INFO Checking if running as non-root
INFO Checking if running inside WSL2
INFO Checking if crc-admin-helper executable is cached
INFO Checking for obsolete admin-helper executable
INFO Checking if running on a supported CPU architecture
INFO Checking minimum RAM requirements
INFO Checking if crc executable symlink exists
INFO Checking if Virtualization is enabled
INFO Checking if KVM is enabled
INFO Checking if libvirt is installed
INFO Checking if user is part of libvirt group
INFO Checking if active user/process is currently part of the libvirt group
INFO Checking if libvirt daemon is running
INFO Checking if a supported libvirt version is installed
INFO Checking if crc-driver-libvirt is installed
INFO Checking if systemd-networkd is running
INFO Checking if NetworkManager is installed
INFO Checking if NetworkManager service is running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
INFO Checking if libvirt 'crc' network is available
INFO Checking if libvirt 'crc' network is active
INFO Loading bundle: crc_libvirt_4.8.12...
INFO Creating CodeReady Containers VM for OpenShift 4.8.12...
INFO Generating new SSH Key pair...
INFO Generating new password for the kubeadmin user
INFO Starting CodeReady Containers VM for OpenShift 4.8.12...
INFO CodeReady Containers instance is running with IP 192.168.130.11
INFO CodeReady Containers VM is running
INFO Updating authorized keys...
INFO Check internal and public DNS query...
INFO Check DNS query from host...
INFO Verifying validity of the kubelet certificates...
INFO Starting OpenShift kubelet service
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Adding user's pull secret to the cluster...
INFO Updating SSH key to machine config resource...
INFO Waiting for user's pull secret part of instance disk...
INFO Adding proxy configuration to the cluster...
INFO Changing the password for the kubeadmin user
INFO Updating cluster ID...
INFO Updating root CA cert to admin-kubeconfig-client-ca configmap...
INFO Starting OpenShift cluster... [waiting for the cluster to stabilize]
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 5 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager, operator-lifecycle-manager-packageserver
INFO 4 operators are progressing: image-registry, kube-apiserver, kube-controller-manager, openshift-controller-manager
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO 2 operators are progressing: image-registry, kube-apiserver
INFO Operator image-registry is progressing
ERRO Cluster is not ready: cluster operators are still not stable after 15m6.225524002s
INFO Waiting for the proxy configuration to be applied...
INFO Adding crc-admin and crc-developer contexts to kubeconfig...
Started the OpenShift cluster.

The server is accessible via web console at:
  https://console-openshift-console.apps-crc.testing

Log in as administrator:
  Username: kubeadmin
  Password: GIq64-qQ2xo-L3nRe-JwE22

Log in as user:
  Username: developer
  Password: developer

Use the 'oc' command line interface:
  $ eval $(crc oc-env)
  $ oc login -u developer https://api.crc.testing:6443

**Comment by gbraad** [comment]
> ERRO Cluster is not ready: cluster operators are still not stable after 15m6.225524002s

please wait a little longer and try to access the console. 15m is long. Are you doing this on a VM?

**Comment by rupeshad81** [comment]
Hi,

Let me check and update you status.


Thanks

**Comment by rupeshad81** [comment]
Hi,

Please let me know, how can we crosscheck the below setting has been done on the node:



crc config set no-proxy .apps.ilocpfnd401.ocp.corp.amdocs.com,.cluster.local,.ilocpfnd401.ocp.corp.amdocs.com,.svc,10.0.0.0/16,10.254.0.0/16,127.0.0.1,172.30.0.0/16,api-int.ilocpfnd401.ocp.corp.amdocs.com,api.ilocpfnd401.ocp.corp.amdocs.com,etcd-0.ilocpfnd401.ocp.corp.amdocs.com,etcd-1.ilocpfnd401.ocp.corp.amdocs.com,etcd-2.ilocpfnd401.ocp.corp.amdocs.com,ilvcos01.corp.amdocs.com,localhost,.corp.amdocs.com,.apps-crc.testing,.crc.testing


From where to check it.

Thanks

**Comment by rupeshad81** [comment]
Hi,

I am not able to find the below details 


crc config set no-proxy .apps.ilocpfnd401.ocp.corp.amdocs.com,.cluster.local,.ilocpfnd401.ocp.corp.amdocs.com,.svc,10.0.0.0/16,10.254.0.0/16,127.0.0.1,172.30.0.0/16,api-int.ilocpfnd401.ocp.corp.amdocs.com,api.ilocpfnd401.ocp.corp.amdocs.com,etcd-0.ilocpfnd401.ocp.corp.amdocs.com,etcd-1.ilocpfnd401.ocp.corp.amdocs.com,etcd-2.ilocpfnd401.ocp.corp.amdocs.com,ilvcos01.corp.amdocs.com,localhost,.corp.amdocs.com,.apps-crc.testing,.crc.testing


[crc@eaas-crc ~]$ ssh -i ~/.crc/machines/crc/id_ecdsa -o StrictHostKeyChecking=no core@$(crc ip)
Red Hat Enterprise Linux CoreOS 48.84.202109100857-0
  Part of OpenShift 4.8, RHCOS is a Kubernetes native operating system
  managed by the Machine Config Operator (`clusteroperator/machine-config`).

WARNING: Direct SSH access to machines is not recommended; instead,
make configuration changes via `machineconfig` objects:
  https://docs.openshift.com/container-platform/4.8/architecture/architecture-rhcos.html

---
[core@crc-scmfr-master-0 ~]$
[core@crc-scmfr-master-0 ~]$
[core@crc-scmfr-master-0 ~]$ sudo systemctl cat crio
# /usr/lib/systemd/system/crio.service
[Unit]
Description=Open Container Initiative Daemon
Documentation=https://github.com/cri-o/cri-o
Requires=crio-wipe.service
Before=kubelet.service
After=network-online.target crio-wipe.service

[Service]
Type=notify
EnvironmentFile=-/etc/sysconfig/crio
EnvironmentFile=-/etc/sysconfig/crio-metrics
EnvironmentFile=-/etc/sysconfig/crio-network
EnvironmentFile=-/etc/sysconfig/crio-storage
Environment=GOTRACEBACK=crash
ExecStart=/usr/bin/crio \
          $CRIO_STORAGE_OPTIONS \
          $CRIO_NETWORK_OPTIONS \
          $CRIO_METRICS_OPTIONS
ExecReload=/bin/kill -s HUP $MAINPID
TasksMax=infinity
LimitNOFILE=1048576
LimitNPROC=1048576
LimitCORE=infinity
OOMScoreAdjust=-999
TimeoutStartSec=0
Restart=on-abnormal

[Install]
WantedBy=multi-user.target

# /etc/systemd/system/crio.service.d/10-mco-default-env.conf
[Service]
EnvironmentFile=/etc/mco/proxy.env

# /etc/systemd/system/crio.service.d/10-mco-default-madv.conf
[Service]
Environment="GODEBUG=x509ignoreCN=0,madvdontneed=1"

# /etc/systemd/system/crio.service.d/10-mco-profile-unix-socket.conf
[Service]
Environment="ENABLE_PROFILE_UNIX_SOCKET=true"

**Comment by praveenkumar** [comment]
```
/etc/systemd/system/crio.service.d/10-mco-default-env.conf 
[Service]
EnvironmentFile=/etc/mco/proxy.env
```

^^ should be part of this file.

**Comment by gbraad** [comment]
> INFO Updating root CA cert to admin-kubeconfig-client-ca configmap...

This was added to the output. Inside the VM you can probably get these `/etc/mco/proxy.env` contents to confirm, but I doubt this is needed.

The VM is slower to start as this is likely running on nested virtualization. Give the startup some time and verify if the OpenShift console can be accessed.

**Comment by rupeshad81** [comment]
Hi,

Thanks.


How can we update /etc/resolv.conf  with same way crc config ?

I wanted to set the below 

ssh -i ~/.crc/machines/crc/id_ecdsa -o StrictHostKeyChecking=no core@$(crc ip) sudo sed -i '/search/s/$/" corp.amdocs.com"/' /etc/resolv.conf


But it is not able to modify the file.


[crc@eaas-crc ~]$ ssh -i ~/.crc/machines/crc/id_ecdsa -o StrictHostKeyChecking=no core@$(crc ip)
Red Hat Enterprise Linux CoreOS 48.84.202109100857-0
  Part of OpenShift 4.8, RHCOS is a Kubernetes native operating system
  managed by the Machine Config Operator (`clusteroperator/machine-config`).

WARNING: Direct SSH access to machines is not recommended; instead,
make configuration changes via `machineconfig` objects:
  https://docs.openshift.com/container-platform/4.8/architecture/architecture-rhcos.html

---
Last login: Mon Oct  4 11:37:04 2021 from 192.168.130.1
[core@crc-scmfr-master-0 ~]$ ping ilvpbg1943
ping: ilvpbg1943: Name or service not known
[core@crc-scmfr-master-0 ~]$ sudo cat /etc/resolv.conf
# Generated by NetworkManager
nameserver 192.168.130.1

**Comment by gbraad** [comment]
> Please let me know, how can we crosscheck the below setting has been done on the node:

Is this a question or additional notes to this issue? Please do not confuse the thread.

**Comment by gbraad** [comment]
> How can we update /etc/resolv.conf with same way crc config ?

You can not

**Comment by rupeshad81** [comment]
Hi,

We want to update the file /etc/resolv.conf  file , so that we can ping the ilvpbg1943

[crc@eaas-crc ~]$ ssh -i ~/.crc/machines/crc/id_ecdsa -o StrictHostKeyChecking=no core@$(crc ip) ping ilvpbg1943
ping: ilvpbg1943: Name or service not known

How can we resolve this ?

**Comment by gbraad** [comment]
> How can we resolve this ?

Start a discussion or a new issue to deal with this.
This issue is considered to be resolved.

**Comment by shawgrey** [comment]
> Hi,
> 
> But in the crc status, Openshift is showing "Degraded"
> 
> ```
> [crc@eaas-crc ~]$ crc status
> CRC VM:          Running
> OpenShift:       Degraded (v4.8.12)
> Disk Usage:      15.15GB of 32.74GB (Inside the CRC VM)
> Cache Usage:     16.06GB
> Cache Directory: /home/crc/.crc/cache
> ```

I was having the same issue but after 10 minutes "Degraded changed to Starting" and is still Starting.
![image](https://github.com/crc-org/crc/assets/111870951/e1dfcf45-b87e-4c46-b173-0b92a12d6caa)

I have tried 
>crc stop
crc delete
crc cleanup
crc setup
crc start --log-level debug

still getting same "Starting"

**Comment by cfergeau** [comment]
Please open a new issue, this issue is almost 2 years old, lots of things have changed in the mean time.
EDIT: I see you've already done this in https://github.com/crc-org/crc/issues/3791
